## Домашнее задание №3
***

### Задача 1
Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки

In [1]:
import numpy as np
from math import factorial

#### Решение
а) среднее арифметическое выборки рассчитывается по формуле:  $\displaystyle M(X)=\frac{1}{n}\sum^n_{i=1}x_i$,\
где $x_i$ - значение случайной величины\
где $n$ - объем выборки

In [2]:
selection = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])
selection

array([100,  80,  75,  77,  89,  33,  45,  25,  65,  17,  30,  24,  57,
        55,  70,  75,  65,  84,  90, 150])

In [3]:
def get_average_value(array):
    return sum(selection)/len(selection)

In [4]:
print(f'Среднее арифметическое = {get_average_value(selection)}')

Среднее арифметическое = 65.3


In [5]:
print(f'Среднее арифметическое, рассчитанное с помощью функции np.mean = {np.mean(selection)}')

Среднее арифметическое, рассчитанное с помощью функции np.mean = 65.3


б) среднее квадратическое отклонение для выборки рассчитывается по формуле: $S=\sqrt{\frac{\displaystyle\sum^n_{i=1}(x_{i}-\bar x)^2}{n - 1}}$,\
среднее квадратичное отклонение показывает, насколько далеко наблюдения могут быть "разбросаны" относительно среднего значения

In [6]:
def get_mean_square_deviation(array):
    square_dev=(array-get_average_value(array))**2
    return (sum(square_dev)/(len(array)-1))**(1/2)

In [7]:
print(f'Среднее квадратичное отклонение = {get_mean_square_deviation(selection): .2f}')

Среднее квадратичное отклонение =  31.62


In [8]:
print(f'Среднее квадратическое отклонение, рассчитанное с помощью функции np.std = {np.std(selection, ddof=1): .2f}')

Среднее квадратическое отклонение, рассчитанное с помощью функции np.std =  31.62


в) смещенная оценка дисперсии для выборки рассчитывается по формуле: $S^2=\frac{\displaystyle\sum^n_{i=1}(x_{i}-\bar x)^2}{n}$,\
несмещенная $S^2=\frac{\displaystyle\sum^n_{i=1}(x_{i}-\bar x)^2}{n-1}$,\
отражает меру разброса данных вокруг средней арифметической

In [9]:
def get_variance(array, no_off=False):
    square_dev=(array-get_average_value(array))**2
    return sum(square_dev)/(len(array)-1) if no_off else sum(square_dev)/len(array)

In [10]:
print(f'Смещенная оценка дисперсии = {get_variance(selection): .2f}\n'
      f'Несмещенная оценка дисперсии = {get_variance(selection, True): .2f}')

Смещенная оценка дисперсии =  950.11
Несмещенная оценка дисперсии =  1000.12


In [11]:
print(f'Смещенная оценка дисперсии, рассчитанная с помощью функции np.var = {np.var(selection): .2f}\n'
     f'Несмещенная оценка дисперсии, рассчитанная с помощью функции np.var = {np.var(selection, ddof=1): .2f}')

Смещенная оценка дисперсии, рассчитанная с помощью функции np.var =  950.11
Несмещенная оценка дисперсии, рассчитанная с помощью функции np.var =  1000.12


***
### Ответ:
**Среднее арифметическое 65,3**\
**Среднее квадратическое отклонение 31,62 (данные в выборке достаточно разбросаны)** \
**Смещенная оценка дисперсии 950,11**\
**Несмещенная оценка дисперсии 1000,12**
***

### Задача 2
В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

#### Решение
Cобытие $A$ "из двух корзин вытащили 3 белых мяча" может произойти следующими способами:  
   1) $A_1$ из первой корзины не вытащили ни одного белого мяча, из второй вытащили 3 белых мяча;  
   2) $A_2$ из первой корзины вытащили 1 белый мяч, из второй вытащили 2 белых мяча;  
   3) $A_3$ из первой корзины вытащили 2 белых мяча, из второй вытащили 1 белый мяч.  
Вероятность события $A$ будет определяться как сумма вероятностей этих событий.  

1. Определим вероятность события $A_1$ (из первой корзины не вытащили ни одного белого мяча, из второй вытащили 3 белых мяча):  
$P(A_1)=\frac{C^2_3}{C^2_8}\cdot\frac{C^3_5\cdot C^1_7}{C^4_{12}}$

In [12]:
def get_combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k))) 

In [13]:
p_A1 = get_combinations(3, 2) / get_combinations(8, 2) * get_combinations(5, 3) * get_combinations(7, 1) / get_combinations(12, 4)
print(f'Вероятность события А1 = {round(p_A1, 4)}')

Вероятность события А1 = 0.0152


2. Определим вероятность события $A_2$ (из первой корзины вытащили 1 белый мяч, из второй вытащили 2 белых мяча):  
$P(A_2)=\frac{C^1_5\cdot C^1_3}{C^2_8}\cdot\frac{C^2_5\cdot C^2_7}{C^4_{12}}$

In [14]:
p_A2 = get_combinations(5, 1) * get_combinations(3, 1) / get_combinations(8, 2) * get_combinations(5, 2) * get_combinations(7, 2) / get_combinations(12, 4)
print(f'Вероятность события А2 = {round(p_A2, 4)}')

Вероятность события А2 = 0.2273


3. Определим вероятность события $A_3$ (из первой корзины вытащили 2 белых мяча, из второй вытащили 1 белый мяч):  
$P(A_3)=\frac{C^2_5}{C^2_8}\cdot\frac{C^1_5\cdot C^3_7}{C^4_{12}}$

In [15]:
p_A3 = get_combinations(5, 2) / get_combinations(8, 2) * get_combinations(5, 1) * get_combinations(7, 3) / get_combinations(12, 4)
print(f'Вероятность события А3 = {round(p_A3, 4)}')

Вероятность события А3 = 0.1263


Общую вероятность события $A$ определим по формуле: $P(A)=P(A_1+A_2+A_3)=P(A_1)+P(A_2)+P(A_3)$

In [16]:
p = p_A1 + p_A2 + p_A3
print(f'Вероятность, что 3 мяча белые = {round(p, 4)}')

Вероятность, что 3 мяча белые = 0.3687


---  
#### Ответ:  
**Вероятность, что 3 мяча белые $\approx 0.3687\approx 36.9\%$**  

---

### Задача 3
На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен:\
a) первым спортсменом\
б) вторым спортсменом\
в) третьим спортсменом?

#### Решение
Для решения задачи воспользуемся формулой:\
$P(A)=P(B_1)\cdot P(B_1|A)+P(B_2)\cdot P(B_2|A)+P(B_3)\cdot P(B_3|A)$\
По условию задачи событие А "выстрел попал в мишень" может наступить только при  наступлении событий:\
$B_1$ "стрелял первый спорстмен",\
$B_2$ "стрелял второй спортсмен",\
$B_3$ "стрелял третий спортсмен",  образующих полную группу событий.\
Вероятности того, что выстрел произведен определенным спортсменом равны, следовательно $P(B_1)=P(B_2)=P(B_3)=\frac{1}{3}$   

In [40]:
PA = 1/3 * 0.9 + 1/3 * 0.8 + 1/3 * 0.6
print(f'Полная вероятность наступления события А = {PA: .4f}')

Полная вероятность наступления события А =  0.7667


Чтобы определить вероятность того, что выстрел произведен конкретным стрелком воспользуемся формулой Байеса:\
$P(B|A)=\frac{P(A|B)\cdot P(B)}{P(A)}$

In [39]:
PB1_A = 1/3 * 0.9 / PA
PB2_A = 1/3 * 0.8 / PA
PB3_A = 1/3 * 0.6 / PA
print(f'P(B_1|A): {PB1_A: .4f}\n'
      f'P(B_2|A): {PB2_A: .4f}\n'
      f'P(B_3|A): {PB3_A: .4f}')

P(B_1|A):  0.3636
P(B_2|A):  0.3232
P(B_3|A):  0.2424


---  
 
#### Ответ:  
а) Вероятность, что мишень поразил первый спортсмен:  $\approx0.3913\approx39.1\%$\
б) Вероятность, что мишень поразил второй спортсмен:  $\approx0.3478\approx34.8\%$\
в) Вероятность, что мишень поразил третий спортсмен:  $\approx0.2609\approx26.1\%$

---

### Задача 4
В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится:\
a) на факультете A\
б) на факультете B\
в) на факультете C?

#### Решение
Решение задачи аналогично решению задачи 3 и будет рассчитываться по формуле:\
$P(A)=P(B_1)\cdot P(B_1|A)+P(B_2)\cdot P(B_2|A)+P(B_3)\cdot P(B_3|A)$\
Событие $A$ - студент сдал первую сессию, может наступить только при  наступлении событий, образующих полную группу:\
$B_1$ - студент учится на факультете А,\
$B_2$ - студент учится на факультете В,\
$B_3$ - студент учится на факультете С

Из условия задачи можно вывести, что:\
$P(B_1)=0,25$\
$P(B_2)=0,25$\
$P(B_3)=0,5$ 

In [38]:
PA = 0.25 * 0.8 + 0.25 * 0.7 + 0.5 * 0.9
print(f'Полная вероятность наступления события А = {PA: .4f}')

Полная вероятность наступления события А =  0.8250


По формуле Байеса  вычислим вероятность того, на каком факультете учится студент, успешно сдавший сессию:\
$P(B|A)=\frac{P(A|B)\cdot P(B)}{P(A)}$

In [31]:
PB1_A = 0.25 * 0.8 / PA
PB2_A = 0.25 * 0.7 / PA
PB3_A = 0.5 * 0.9 / PA
print(f'P(B_1|A): {PB1_A: .4f}\n'
      f'P(B_2|A): {PB2_A: .4f}\n'
      f'P(B_3|A): {PB3_A: .4f}')

P(B_1|A):  0.2424
P(B_2|A):  0.2121
P(B_3|A):  0.5455


---  
 
#### Ответ:  
а) Вероятность того, студент успешно сдавший первую сессию учится на факультете А:  $\approx0.2424\approx24.2\%$\
б) Вероятность того, студент успешно сдавший первую сессию учится на факультете В:  $\approx0.2121\approx21.2\%$\
в) Вероятность того, студент успешно сдавший первую сессию учится на факультете С:  $\approx0.5455\approx54.6\%$

---

### Задача 5
Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя:\
а) все детали\
б) только две детали\
в) хотя бы одна деталь\
г) от одной до двух деталей?

#### Решение
Определим возможные события:\
а) Событие $A_1$ - вышла из строя первая деталь, $P(A_1)=0.1$;\
б) Событие $\bar A_1$ - первая деталь не вышла из строя (противоположное событию $A_1$), $P(\bar A_1)=1-P(A_1)=1-0.1=0,9$;\
в) Событие $A_2$ - вышла из строя вторая деталь, $P(A_2)=0.2$;\
г) Событие $\bar A_2$ - вторая деталь не вышла из строя (противоположное событию $A_2$), $P(\bar A_2)=1-P(A_2)=1-0.2=0,8$;\
д) Событие $A_3$ - вышла из строя третья деталь, $P(A_3)=0.25$;\
е) Событие $\bar A_3$ - третья деталь не вышла из строя (противоположное событию $A_3$), $P(\bar A_3)=1-P(A_3)=1-0.25=0,75$. 

а) Вероятность события "вышли из строя все детали",  определим путем произведения вероятностей событий.  
$P(3)=P(A_1)\cdot P(A_2)\cdot P(A_3)=0.1\cdot 0.2\cdot 0.25$

In [34]:
P3 = 0.1 * 0.2 * 0.25
print(f'Вероятность того, что вышли из строя все детали = {P3: .3f}')

Вероятность того, что вышли из строя все детали =  0.005


б) Cобытие  "вышли из строя две детали", произойдет в результате следующих комбинаций событий:  
   1. Произошли события $A_1$ **и** $A_2$ **и** $\bar A_3$;  
   2. Произошли события $A_1$ **и** $\bar A_2$ **и** $A_3$;  
   3. Произошли события $\bar A_1$ **и** $A_2$ **и** $A_3$.  
   
Вероятность события определим суммой произведений комбинаций вероятностей этих событий

In [37]:
P2 = (0.1 * 0.2 * 0.75) + (0.1 * 0.8 * 0.25) + (0.9 * 0.2 * 0.25)
print(f'Вероятность того, что вышли из строя две детали = {P2: .2f}')

Вероятность того, что вышли из две детали =  0.08


в) Вероятность события "выйдет из строя хотя бы одна деталь" рассчитаем как противоположное событию "не вышло из строя ни одной детали"


In [43]:
P1 = 1 - (0.9 * 0.8 * 0.75)
print(f'Вероятность того, что выйдет из строя хотя бы одна деталь = {P1: .2f}')

Вероятность того, что выйдет из строя хотя бы одна деталь =  0.46


г) Событие "из строя выйдут от одной до двух деталей" произойдет при наступлении события "из строя выйдет одна деталь" или "из строя выйдут две детали". Его вероятность будет равна сумме вероятностей этих событий: $P(1,2)=P(1)+P(2)$

События "не выйдет из строя ни одной детали", "из строя выйдет одна деталь", "из строя выйдет 2 детали", "из строя выйдут все детали" образуют полную группу событий, сумма вероятностей наступления этих событий равна 1.  
$P(0)+P(1)+P(2)+P(3)=1\\0.54+P(1)+0.08+0.005=1\\P(1)=0.375$


In [46]:
P1_2 = 0.375 + 0.08
print(f'Вероятность выхода из строя от одной до двух деталей = {P1_2: .3f}')

Вероятность выхода из строя от одной до двух деталей =  0.455


---  
 
#### Ответ:  
а) Вероятность того, что в первый месяц из строя вышли все детали:  $\approx0.005\approx0.5\%$\
б) Вероятность того, что в первый месяц из строя вышли только две детали:  $\approx0.08\approx8.0\%$\
в) Вероятность того, что в первый месяц из строя вышла хотя бы одна деталь:  $\approx0.46\approx46.0\%$\
г) Вероятность того, что в первый месяц из строя выйдут от одной до двух деталей: $\approx0.455\approx45.5\%$

---